CMake/test_labels.cmake

```diff
@@ -1,7 +1,9 @@
 function(ADD_TEST_LABELS TEST_NAME TEST_LABELS)
   set(TEST_LABELS_TEMP "")
-  if (DEFINED TEST_LABELS_${TEST_NAME}_${QMC_CUDA}_${QMC_COMPLEX}_${QMC_MIXED_PRECISION})
-    set(TEST_LABELS_TEMP TEST_LABELS_${${TEST_NAME}_${QMC_CUDA}_${QMC_COMPLEX}_${QMC_MIXED_PRECISION}})
+  file(TIMESTAMP ${qmcpack_SOURCE_DIR}/tests/scripts/test_labels.py TEST_LABELS_TIMESTAMP)
```

Comment:
Growing the number of `TEST_LABELS_XXX` variables over time is not good.
Can we just cache TEST_LABELS_TIMESTAMP and compare at every include of "CMake/test_labels.cmake"?
I mean move "file(TIMESTAMP" before the function. Otherwise, it runs at every function call.

Reply:
I moved `file(TIMESTAMP...` to CMakeLists.txt so it is only called once. If it changes, any cached variables names starting with `TEST_LABELS_` are unset.